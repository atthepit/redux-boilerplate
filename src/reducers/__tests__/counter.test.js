import counter from '../counter'
import * as actions from '../../actions/counter'

describe('Counter reducer', () => {
  test('undefined action', () => {
    const previousState = undefined
    const action = {
      type: 'JIBBERISH',
    }

    expect(
      counter(previousState, action)
    ).toMatchSnapshot()
  })

  test('Test increment counter', () => {
    const previousState = 0
    const action = actions.increment()

    expect(
      counter(previousState, action)
    ).toMatchSnapshot()
  })

  test('Test decrement counter', () => {
    const previousState = 1
    const action = actions.decrement()

    expect(
      counter(previousState, action)
    ).toMatchSnapshot()
  })

})
