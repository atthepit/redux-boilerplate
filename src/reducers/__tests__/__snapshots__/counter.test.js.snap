exports[`Counter reducer Test decrement counter 1`] = `0`;

exports[`Counter reducer Test increment counter 1`] = `1`;

exports[`Counter reducer undefined action 1`] = `0`;
